### PR TITLE
Velocity Backend injection / Velocity 1.1.0

### PR DESCRIPTION
**Pull request by creeper123123321**
Implements #1550
~~Depends on #1505, #1719 and VelocityPowered/Velocity#294~~ All merged

### CODE REVIEW

File: velocity/src/main/java/us/myles/ViaVersion/velocity/handlers/backend/VelocityBackendDecodeHandler.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        if (cause instanceof CancelDecoderException) return;
+        super.exceptionCaught(ctx, cause);
+    }
+
+    @Override
+    public void channelInactive(ChannelHandlerContext ctx) throws Exception {
+        super.channelInactive(ctx);
+
+        ProtocolInfo info = user.get(ProtocolInfo.class);
+        if (info.getUuid() != null) {
```
Comment (creeper123123321) [inline velocity/src/main/java/us/myles/ViaVersion/velocity/handlers/backend/VelocityBackendDecodeHandler.java:91]: Shouldn't use UUID

**Comment by creeper123123321** [comment]
Needs some changes for handling backend UserConnection kicking

**Comment by creeper123123321** [comment]
Todo:
- [ ] override UserConnection

**Comment by creeper123123321** [comment]
Moved into #1505

**Comment by creeper123123321** [comment]
Needs fixing compression on backend compression

**Comment by creeper123123321** [comment]
Needs fixing: random disconnects

**Comment by Justsnoopy30** [comment]
Status on this PR?

**Comment by Justsnoopy30** [comment]
Could someone take a look at this to try to finish it up please? I still cannot use ViaVersion on Velocity 1.1.0, or on the backend servers behind Velocity 1.1.0.

**Comment by creeper123123321** [comment]
Bug needing fix: where's block tags?

**Comment by creeper123123321** [comment]
Some bugs were fixed by https://github.com/VelocityPowered/Velocity/commit/e57c6d00c00b13b827026d4822a1f9d5e09bf679

**Comment by creeper123123321** [comment]
BUG:
```
io.netty.handler.codec.CorruptedFrameException: Did not read full packet for class com.velocitypowered.proxy.protocol.packet.Disconnect Direction CLIENTBOUND Protocol 1.7.6 State LOGIN ID 0```

**Comment by creeper123123321** [comment]
^ bug fixed in ViaVersion/ViaRewind#184

**Comment by johncsuti** [comment]
Can this be merged please?

**Comment by Xernium** [comment]
@astei is right on this. If we merge this we break support for the old stable releases of velocity. This will be merged after velocity 1.1.0 is released, or after we have support for both old and new velocity versions.

**Comment by johncsuti** [comment]
does anyone have a version of this that i can use i down care about the old version of velocity

**Comment by astei** [comment]
Compile this PR yourself.

**Comment by johncsuti** [comment]
I don’t know how to. Otherwise I would have.

**Comment by astei** [comment]
Velocity 1.1.0 is now released. This can be merged in now.

**Pull request merged**